starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -169,12 +175,12 @@ def from_python(self, cairo_type, name, value) -> CairoData:
             return [value & ((1 << 128) - 1), value >> 128]
 
         if not isinstance(value, dict):
-            raise TypeError(f"Expected {name} to be a dict.")
+            raise InvalidDataException(f"Expected {name} to be a dict.")
```

Comment:
This is a breaking change

Reply:
You're right, I'll return to the old TypeError here because it is catched in the CairoSerializer.from_python

Reply:
Can't we just throw an exception inheriting a proper type here? That's way it's not breaking.

Reply:
It is CairoSerializerException and I would like avoid to use it outside CairoSerializer